src/nvim/buffer.c

```diff
@@ -4528,16 +4528,20 @@ chk_modeline (
           e = s + 4;
         else
           e = s + 3;
-        vers = getdigits_int(&e);
+        if (getdigits_safe(&e, &vers) < 0) {
```

Comment:
This is a change in behavior from Vim.  This causes nvim to skip the modeline entirely, where as Vim ends up getting `vers = -1` since it downcasts the long to an int and then uses that in the comparison on line 4539.  The comparison succeeds and the modeline is set.

Now, one can question whether that's the correct behavior for Vim (and I would suggest bringing it up there), but that still leaves open how nvim should behave.  I'm in favor of ignoring the modeline, since I think it's better to be overcautious with modelines, but we should also be aware of and intentional about places where we deviate in behavior from Vim.